tests/gradients/test_parameter_shift.py

```diff
@@ -339,38 +336,38 @@ def grad_recipe(self):
         grad = fn(dev.batch_execute(tapes))
         assert np.allclose(grad, -np.sin(x))

```

Comment:
@dwierichs is it possible to use a mock or a spy here to ensure that `grad_recipe` is being used?

Reply:
I included a monkeypatch that removes `parameter_frequencies` for `qml.RX`, so that we are sure that the recipe only can be known from `grad_recipe` and not from `parameter_frequencies` (or indirectly via the default `parameter_frequencies` from the `generator`)

Reply:
Nice!